### ISSUE TITLE
Use another K8S API specific endpoint with gofabric8

### ISSUE THREAD

**Issue by ThomasSchweizer** [fabric8io/gofabric8#127]
Hello,
We are using Rancher for deploying our K8S cluster. Basically Rancher is deploying Kubernetes with only HTTP port 80 exposed so we have to access the API on _http://kubernetes:80_

When connecting to Fabric8 Webconsole I have a lot of errors in the fabric8 console logs because fabric8 is trying to call the API at _https://kubernetes:443_

I can access the K8S API in HTTPS but only over Rancher proxy
Basically to access the K8S API I have to connect to : **https://rancher.mycompany/r/projects/1a012/kubernetes/api/v1**

Any idea how can I set this during the gofabric8 deployment process?

Thank you for your advices :)
Best regards, Thomas

**Comment by jstrachan** [comment]
Note that the https://kubernetes:443 is using DNS to resolve the kubernetes service name "kubernetes" to the internal kubernetes service port IP. 

I wonder if the issue is that DNS isn't enabled so that kubernetes DNS name doesn't get resolved inside the fabric8 console pod? 

e.g. if you do

```
kubectl exec -it fabric8-1234 bash

$ curl -k https://kubernetes:443/
{
  "paths": [
    "/api",
    "/api/v1",
    "/apis",
    "/controllers",
    "/healthz",
    "/healthz/ping",
    "/healthz/ready",
    "/metrics",
    "/oapi",
    "/oapi/v1",
    "/swaggerapi/"
  ]
}
```

and you should get some JSON back as above. If that doesn't work it sounds like the kubernetes service DNS resolution isn't working?

**Comment by jstrachan** [comment]
incidentally what version of kubernetes is being used? I thought kubernetes was always exposed on 443?

As a temporary workaround you could edit the ReplicationController `fabric8` via `kubectl edit rc fabric8` to change this line:
```
      containers:
        - name: fabric8
          image: 'fabric8/fabric8-console:2.2.174'
          command:
            - /kuisp
            - '-p'
            - '9090'
            - '-c'
            - /site/osconsole/config.basic.js.tmpl=/site/osconsole/config.js
            - '--bearer-token=/var/run/secrets/kubernetes.io/serviceaccount/token'
            - '--skip-cert-validation=true'
            - '--default-page=/index.html'
            - '--max-age=24h'
            - '-s'
            - '/k8s/=https://kubernetes:443/'
```

to be 

```
            - '/k8s/=http://kubernetes:80/'
```

I'd like to figure out why 443 isn't allowed though. If we have to we can try figure out how to make that URL more easily configurable

**Comment by ThomasSchweizer** [comment]
Hello @jstrachan 

Thanks for the advice I will try this :)

K8S seems to be on v1.2.4
Server Version: version.Info{Major:"1", Minor:"2+", GitVersion:"v1.2.4-rancher9", GitCommit:"4ed40fd4eaa5717b11406f34206308fdf6ad4651", GitTreeState:"clean", BuildDate:"", GoVersion:"", Compiler:"", Platform:""}

I opened an issue with them to see why : https://github.com/rancher/rancher/issues/5918

Will let you know how it is working

Regards;
Thomas

**Comment by jstrachan** [comment]
@jimmidyson any idea on whether or not the kubernetes service should be on 443 or if it could be on either? I guess we might need to grok the kubernetes service on startup to see what port its listening on in kuisp or something

**Comment by jimmidyson** [comment]
It can be on either I guess, although almost every install is secured now by default. We can probably check the port somehow /SRV records?) without needing to query API server.

**Comment by ThomasSchweizer** [comment]
@jstrachan 

With your advice I changed the RC settings and recreated the pod

The panel is loading perfectly now but it seems like the API is heavily modified as when I try to add a project in fabric8 console it goes nowhere and I get a 404 error in the browser developer mode while the fabric8 console is trying to load http://<consoleip>/k8s/api/v1/namespaces/default/ingresses

{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "the server could not find the requested resource",
  "reason": "NotFound",
  "details": {},
  "code": 404
}

I assume I'll maybe deploy a  vanilla K8S for fixing all this :)

**Comment by jstrachan** [comment]
@ThomasSchweizer the console not being able to load some resources isn't the end of the world TBH; its just a warning in the JavaScript log. Does the rest of the web console seem to work OK

**Comment by ThomasSchweizer** [comment]
@jstrachan 
Yes It finally worked but in the backend it is throwing a lot of errors since the API (even the K8S one) seems to be heavily modified.

I consider that Fabric8 issue is now ok on my side, I'll pursue with Rancher team to check why it is going sideways.

Thanks again for your help :) 

Regards,
Thomas
